server/src/main/java/org/elasticsearch/search/aggregations/bucket/histogram/DateHistogramAggregator.java

```diff
@@ -428,7 +428,8 @@ static class FromDateRange extends AdaptingAggregator implements SizedBucketAggr
         protected InternalAggregation adapt(InternalAggregation delegateResult) {
             InternalDateRange range = (InternalDateRange) delegateResult;
             List<InternalDateHistogram.Bucket> buckets = new ArrayList<>(range.getBuckets().size());
-            for (InternalDateRange.Bucket rangeBucket : range.getBuckets()) {
+
+            range.getBuckets().stream().forEach(rangeBucket -> {
```

Comment:
Please add a comment referencing the JDK bug and the fact this needs to stay a stream foreach to avoid the bug

Reply:
Good catch, I thought about doing and I forgot.